### PR TITLE
Fixed test_get_transceiver_info testcase for ZR SFPs

### PR DESCRIPTION
**Pull request by mannytaheri**
### Description of PR
Fixed test_get_transceiver_info testcase for ZR SFPs.

Summary:
Fixes # (issue)

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [ x] Test case(new/improvement)

### Back port request
- [ ] 201911
- [ ] 202012
- [x] 202205

#### What is the motivation for this PR?
To make test_get_transceiver_info pass if SFP is ZR.
test_get_transceiver_info testcase fails for a ZR SFP with the message:
Failed: Transceiver info contains unexpected field: 
'supported_min_laser_freq', 'supported_max_tx_power', 'supported_min_tx_power', 'supported_max_laser_freq'

#### How did you do it?
Added the above keys to tranceiver info keys if SFP is a ZR QSFP-DD

#### How did you verify/test it?
Tested test_get_transceiver_info testcase on a multi cards multi ASICs chassis

### CODE REVIEW

**Comment by wangxin** [comment]
The [pre-commit](http://pre-commit.com/) check detected issues in the files touched by this pull request.
The detected issues may be old or new. For new issues, please try to fix them.

For old issues, it is not mandatory to fix them because they were not caused by this change. It is unfair to blame
author of this pull request. But if you can take extra effort to fix the old issues as well, that would be great!

Detailed pre-commit check results:
<samp>trim trailing whitespace.................................................Passed<br/>fix end of files.........................................................Passed<br/>check yaml...........................................(no files to check)Skipped<br/>check for added large files..............................................Passed<br/>check python ast.........................................................Passed<br/>flake8...................................................................Failed<br/>- hook id: flake8<br/>- exit code: 1<br/><br/>tests/platform_tests/api/test_sfp.py:11:1: F401 'tests.common.fixtures.conn_graph_facts.conn_graph_facts' imported but unused<br/>tests/platform_tests/api/test_sfp.py:12:1: F401 'tests.common.fixtures.duthost_utils.shutdown_ebgp' imported but unused<br/>tests/platform_tests/api/test_sfp.py:22:19: F821 undefined name 'basestring'<br/>tests/platform_tests/api/test_sfp.py:33:1: E302 expected 2 blank lines, found 1<br/>tests/platform_tests/api/test_sfp.py:34:80: F811 redefinition of unused 'conn_graph_facts' from line 11<br/>tests/platform_tests/api/test_sfp.py:34:98: F811 redefinition of unused 'shutdown_ebgp' from line 12<br/>tests/platform_tests/api/test_sfp.py:46:59: E502 the backslash is redundant between brackets<br/>tests/platform_tests/api/test_sfp.py:47:9: E128 continuation line under-indented for visual indent<br/>tests/platform_tests/api/test_sfp.py:53:71: E502 the backslash is redundant between brackets<br/>tests/platform_tests/api/test_sfp.py:54:49: E127 continuation line over-indented for visual indent<br/>tests/platform_tests/api/test_sfp.py:54:80: E502 the backslash is redundant between brackets<br/>...<br/>[truncated extra lines, please run pre-commit locally to view full check results]<br/></samp>

To run the pre-commit checks locally, you can follow below steps:
1. Ensure that default python is python3. In sonic-mgmt docker container, default python is python2. You can run
   the check by activating the python3 virtual environment in sonic-mgmt docker container or outside of sonic-mgmt
   docker container.
2. Ensure that the `pre-commit` package is installed:
```
sudo pip install pre-commit
```
3. Go to repository root folder
4. Install the pre-commit hooks:
```
pre-commit install
```
5. Use pre-commit to check staged file:
```
pre-commit
```
6. Alternatively, you can check committed files using:
```
pre-commit run --from-ref <commit_id> --to-ref <commit_id>
```

**Comment by wangxin** [comment]
The [pre-commit](http://pre-commit.com/) check detected issues in the files touched by this pull request.
The detected issues may be old or new. For new issues, please try to fix them.

For old issues, it is not mandatory to fix them because they were not caused by this change. It is unfair to blame
author of this pull request. But if you can take extra effort to fix the old issues as well, that would be great!

Detailed pre-commit check results:
<samp>trim trailing whitespace.................................................Passed<br/>fix end of files.........................................................Passed<br/>check yaml...........................................(no files to check)Skipped<br/>check for added large files..............................................Passed<br/>check python ast.........................................................Passed<br/>flake8...................................................................Failed<br/>- hook id: flake8<br/>- exit code: 1<br/><br/>tests/platform_tests/api/test_sfp.py:11:1: F401 'tests.common.fixtures.conn_graph_facts.conn_graph_facts' imported but unused<br/>tests/platform_tests/api/test_sfp.py:12:1: F401 'tests.common.fixtures.duthost_utils.shutdown_ebgp' imported but unused<br/>tests/platform_tests/api/test_sfp.py:22:19: F821 undefined name 'basestring'<br/>tests/platform_tests/api/test_sfp.py:33:1: E302 expected 2 blank lines, found 1<br/>tests/platform_tests/api/test_sfp.py:34:80: F811 redefinition of unused 'conn_graph_facts' from line 11<br/>tests/platform_tests/api/test_sfp.py:34:98: F811 redefinition of unused 'shutdown_ebgp' from line 12<br/>tests/platform_tests/api/test_sfp.py:46:59: E502 the backslash is redundant between brackets<br/>tests/platform_tests/api/test_sfp.py:47:9: E128 continuation line under-indented for visual indent<br/>tests/platform_tests/api/test_sfp.py:53:71: E502 the backslash is redundant between brackets<br/>tests/platform_tests/api/test_sfp.py:54:49: E127 continuation line over-indented for visual indent<br/>tests/platform_tests/api/test_sfp.py:54:80: E502 the backslash is redundant between brackets<br/>...<br/>[truncated extra lines, please run pre-commit locally to view full check results]<br/></samp>

To run the pre-commit checks locally, you can follow below steps:
1. Ensure that default python is python3. In sonic-mgmt docker container, default python is python2. You can run
   the check by activating the python3 virtual environment in sonic-mgmt docker container or outside of sonic-mgmt
   docker container.
2. Ensure that the `pre-commit` package is installed:
```
sudo pip install pre-commit
```
3. Go to repository root folder
4. Install the pre-commit hooks:
```
pre-commit install
```
5. Use pre-commit to check staged file:
```
pre-commit
```
6. Alternatively, you can check committed files using:
```
pre-commit run --from-ref <commit_id> --to-ref <commit_id>
```

File: tests/platform_tests/api/test_sfp.py [hunk 1]
```diff
@@ -376,14 +383,14 @@ def test_get_transceiver_info(self, duthosts, enum_rand_one_per_hwsku_hostname,
...
                             key if key != 'vendor_rev' else 'hardware_rev' for key in self.EXPECTED_XCVR_INFO_KEYS]
-                        # self.EXPECTED_XCVR_INFO_KEYS = EXPECTED_XCVR_INFO_KEYS
                     else:
-
-                        if info_dict["type_abbrv_name"] in ["QSFP-DD", "OSFP-8X"]:
-                            UPDATED_EXPECTED_XCVR_INFO_KEYS = self.EXPECTED_XCVR_INFO_KEYS + \
-                                                           self.EXPECTED_XCVR_NEW_QSFP_DD_OSFP_INFO_KEYS + \
-                                                           ["active_apsel_hostlane{}".format(n)
-                                                            for n in range(1, info_dict['host_lane_count'] + 1)]
+                        if info_dict["type_abbrv_name"] == "QSFP-DD":
```
Comment (prgeor) [inline tests/platform_tests/api/test_sfp.py:387]: @mannytaheri this check is removing the previous check supporting OSFP modules. Can you fix this for both QSFP-DD and OSFP?

File: tests/platform_tests/api/test_sfp.py [hunk 2]
```diff
@@ -376,14 +383,14 @@ def test_get_transceiver_info(self, duthosts, enum_rand_one_per_hwsku_hostname,
...
                             key if key != 'vendor_rev' else 'hardware_rev' for key in self.EXPECTED_XCVR_INFO_KEYS]
-                        # self.EXPECTED_XCVR_INFO_KEYS = EXPECTED_XCVR_INFO_KEYS
                     else:
-
                         if info_dict["type_abbrv_name"] in ["QSFP-DD", "OSFP-8X"]:
-                            UPDATED_EXPECTED_XCVR_INFO_KEYS = self.EXPECTED_XCVR_INFO_KEYS + \
-                                                           self.EXPECTED_XCVR_NEW_QSFP_DD_OSFP_INFO_KEYS + \
-                                                           ["active_apsel_hostlane{}".format(n)
-                                                            for n in range(1, info_dict['host_lane_count'] + 1)]
+                            XCVR_INFO_KEYS = self.EXPECTED_XCVR_INFO_KEYS + self.EXPECTED_XCVR_NEW_QSFP_DD_INFO_KEYS + [
```
[the remark this answers is not in the document]
Comment (mannytaheri) [reply tests/platform_tests/api/test_sfp.py]: Fixed

**Review by prgeor — APPROVED** [review]

**Comment by judyjoseph** [comment]
@mannytaheri  can you get the EasyCLA fixed ?

**Pull request merged**